packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -340,6 +343,8 @@ async function processBlogSourceFile(
     id: slug,
     metadata: {
       permalink,
+      baseBlogPermalink,
+      baseBlogTitle,
```

Comment:
that feels awkward to add those 2 attributes here, as they are global to the blog and not really blog post metadata

We'd rather create a new bundle/prop to inject it in all blog post routes instead of adding this here:

```js
          addRoute({
            path: metadata.permalink,
            component: blogPostComponent,
            exact: true,
            modules: {
              sidebar: aliasedSource(sidebarProp),
              content: metadata.source,
              // add blogMetadata bundle/prop here?
            },
          });
```

Reply:
There doesn't appear to be a non-disruptive way to just inject props here as suggested.  The only mechanism I can spot is using the [`createData`](https://docusaurus.io/docs/api/plugin-methods/lifecycle-apis#createData) API - is that what you had in mind?

Reply:
Implemented